app/container-imp.ts

```diff
@@ -67,12 +67,12 @@ export class AppContainer implements Container {
       ])
       const state: BCState = {
         ...initialState,
-        loginAttempt,
-        preferences,
-        migration,
-        tours,
-        onboarding,
-        dismissPersonCredentialOffer: personCredOfferDissmissed,
+        loginAttempt: {...initialState.loginAttempt, ...loginAttempt},
```

Comment:
Using `{...}` to copy an object in JavaScript is good for a basic copy and works in simple cases. But, it only copies the first layer deeply. If you have objects or arrays inside, they're still linked to the original, which may not be what you want for a complete deep copy.